tap/parser.py

```diff
@@ -41,20 +42,39 @@ class Parser(object):
 
     TAP_MINIMUM_DECLARED_VERSION = 13
 
-    def parse_file(self, filename):
-        """Parse a TAP file and determine what each line in the file is.
+    def parse(self, fh):
```

Comment:
The flow of methods in this project is to put the highest level abstraction at the top and the more detailed methods below. `parse_file`, `parse_stdin`, and `parse_text` all make use of the lower level `parse` method. Could you update the methods to come in that order (with the three high level methods in alphabetical order and `parse` below the three)?